lib/dynamoid/persistence.rb

```diff
@@ -152,6 +152,17 @@ def undump_field(value, options)
         end
       end
 
+      def undump_set(options, value)
+        case options[:item_type]
+        when :integer
+          value.map { |v| Integer(v) }.to_set
+        when :number
+          value.map { |v| BigDecimal.new(v.to_s).to_f }.to_set
```

Comment:
Default behavior now is to use `BigDecimal` in order to store set of numbers.
You propose to add new value `number` and to convert to Float. I don't understand why.

For numeric set if you save integer values `{1, 2, 3}`, you will convert them to float at loading `{1.0, 2.0, 3.0}`. It would be surprise to user to get float when it stores set of **numbers**

Anyway, I would like to see proposed changes either in PR's description or issue (to discuss them before implementation)

Reply:
I think we should continue to have `:number` map to `BigDecimal`.  If you want to support float, make it a new type:
```
        when :integer
          value.map { |v| Integer(v) }.to_set
        when :number
          value.map { |v| BigDecimal.new(v.to_s) }.to_set
        when :float
          value.map { |v| v.to_s.to_f }.to_set
```

Reply:
Done